### DISCUSSION TITLE
What IS face clustering?

### DISCUSSION THREAD

**Discussion by 31337-4554551n** [nextcloud/recognize#1162] [Q&A] [answered]
I am not able to find information on what clustering is/does?
I see that locations and objects get recognized only, but faces get recognized and clustered.
What does this mean?

**Comment by marcelklehr** [comment, marked as answer]
Hey @31337-4554551n 
So, faces in photos first need to get recognized, so that the software knows there's a face in that picture, where in the picture it is, and what the face looks like (this is expressed as a vector). To aggregate similar faces in groups, the software searches for clusters in the vector space of face vectors. If some vectors are close enough together (simplifying here), then they will be put into the same cluster and appear as photos of one person in the user interface. If you add new photos and there's a face whose vector is close to an existing cluster's vectors, it will be put into the same cluster. This process of figuring out which group a face might belong to is called clustering.

**Reply by 31337-4554551n** [reply]
> Hey @31337-4554551n So, faces in photos first need to get recognized, so that the software knows there's a face in that picture, where in the picture it is, and what the face looks like (this is expressed as a vector). To aggregate similar faces in groups, the software searches for clusters in the vector space of face vectors. If some vectors are close enough together (simplifying here), then they will be put into the same cluster and appear as photos of one person in the user interface. If you add new photos and there's a face whose vector is close to an existing cluster's vectors, it will be put into the same cluster. This process of figuring out which group a face might belong to is called clustering.

Thank you so much for getting back to me. So I get that now, it groups faces into albumns or what have you. But it doesn't need to do that with objects, it'll just recognise and tag them. So you can search for "table" but wouldn't need to find an albumn of identical tables. Am I understanding it right?

**Reply by marcelklehr** [reply]
Yep, that's it :)

**Reply by 31337-4554551n** [reply]
Thank you so much :)
